docs/config/development/feature-from-scratch/index.rst

```diff
@@ -0,0 +1,385 @@
+===============================
+Creating a Feature from Scratch
+===============================
+
+This guide will outline the process of creating a new feature from the beginning.
+
+If you haven't already, please read the
+:doc:`Config Development Introduction </config/development/introduction>` &
+:doc:`Creating A Pack From Scratch </config/development/pack-from-scratch/index>`
+for more information before continuing.
+
+For a more detailed and in-depth guide about creating a new feature from scratch, please read this unofficial
+development guide, `Feature Config <https://terra.atr.sh/#/page/feature%20config>`__.
+
+Setting up a New Feature
+========================
+
+`PROCEDURE`
+
+1. Add feature stage(s)
+-----------------------
+
+.. card::
+
+    Your pack will need feature stages in order to generate features.
+
+    Open your pack manifest in your :ref:`editor of choice <editor>`.
+
+    Add the ``generation-stage-feature`` addon as a dependency, using versions ``1.+``.
+
+    This addon will allow us to create new feature stages for features within the pack manifest.
+
+    .. code-block:: yaml
+        :caption: pack.yml
+        :linenos:
+        :emphasize-lines: 9
+
+        id: YOUR_PACK_ID
+
+        version: 0.1.0
+
+        addons:
+          language-yaml: "1.+"
+          chunk-generator-noise-3d: "1.+"
+          ...
+          generation-stage-feature: "1.+"
+
+    Add the highlighted lines below to your pack manifest to create these feature stages.
+
+    .. code-block:: yaml
+        :caption: pack.yml
+        :linenos:
+        :emphasize-lines: 5-10
+
+        id: YOUR_PACK_ID
+
+        ...
+
+        stages:
+          - id: trees
+            type: FEATURE
+
+          - id: landforms
+            type: FEATURE
+
+    We'll only be using the ``trees`` stage for this guide.
+
+    .. tip::
+
+        The feature stage ids can be named to your liking and feature stages will generate in order from top to bottom.
+
+2. Create your structure file
+-----------------------------
+
+.. card::
+
+    Structure files can either be dynamic TerraScript ``.tesf`` or static schematics ``.schem`` files.
+
+    .. tab-set::
+
+            .. tab-item:: TerraScript
+
+                TerraScript files consist of :doc:`TerraScript </config/documentation/terrascript/index>` language
+                that typically generate structures procedurally and can create unique complex structure layouts.
+
+                1. Add the ``structure-terrascript-loader`` addon to the pack manifest, using versions ``1.+``
+
+                2. Create a blank ``.tesf`` file.
+
+                3. Add TerraScript within the .tesf file to generate the structure.
```

Comment:
If you have a sample script, linking / including it would be excellent (if you haven't already later on - I haven't read the rest of the guide yet)